### PR TITLE
add `project delete` command

### LINKED ISSUE #40: Add `project delete` command

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: cmd/commands/common.go [hunk 1]
```diff
@@ -73,8 +74,12 @@ var (
 		return r, repofs, nil
 	}
 
-	glob = func(fs fs.FS, pattern string) ([]string, error) {
-		return billyUtils.Glob(fs, pattern)
+	removeAll = billyUtils.RemoveAll
+
+	glob = billyUtils.Glob
+
+	writeFile = func(fs billy.Basic, filename string, data []byte) error {
```
Comment (roi-codefresh) [inline cmd/commands/common.go:81]: I think that this way of assigning a function to a var to later replace in tests should be a last resort. It's really hard to manage and we could end up with a file with tons of unrelated functions assigned to vars.
For example - `writeFile` `glob` `removeAll` should all be just tested using a `memfs`. It's easier and cleaner to do than to replace the functions before and after every test.

File: cmd/commands/project.go [hunk 2]
```diff
@@ -400,26 +379,113 @@ var getProjectInfoFromFile = func(fs fs.FS, name string) (*argocdv1alpha1.AppPro
...
+# and have a bootstrapped GitOps repository, and provide them using:
+	
+		export GIT_TOKEN=<token>
+		export GIT_REPO=<repo_url>
+
+# or with the flags:
+	
+		--token <token> --repo <repo_url>
+		
+# Get list of installed applications in a specifc project
```
Comment (roi-codefresh) [inline cmd/commands/project.go:424]: description is not updated

File: cmd/commands/project.go [hunk 3]
```diff
@@ -400,26 +379,113 @@ var getProjectInfoFromFile = func(fs fs.FS, name string) (*argocdv1alpha1.AppPro
...
+			return fmt.Errorf("failed to read overlays directory '%s': %w", appOverlaysDir, err)
+		}
+
+		appDir := filepath.Dir(appOverlaysDir)
+		appName := filepath.Base(appDir)
+		var dirToRemove string
+		var msg string
+		if len(allOverlays) == 1 {
+			dirToRemove = appDir
+			msg = fmt.Sprintf("Deleting app '%s'", appName)
```
Comment (roi-codefresh) [inline cmd/commands/project.go:467]: why not just log it here?

**Review by roi-codefresh — CHANGES_REQUESTED** [review]

File: cmd/commands/common.go (hunk 1 above)
Comment (noam-codefresh) [reply cmd/commands/common.go:81]: Looking at the current implementation of `billyUtils.Glob` and `billyUtils.RemoveAll`, they seem like they will not be easily mocked by a fsmock.
I agree that WriteFile is much simpler, and we even had our own implementation of it before we noticed billyUtils, so we can easily revert to it.

Comment (roi-codefresh) [reply cmd/commands/common.go:81]: My point is - we don't want to mock `Glob` and `RemoveAll`, let them do their job, just prepare a `memfs` for your tests with the files you need and later assert on the memfs for the expected changes

Comment (roi-codefresh) [reply cmd/commands/common.go:81]: I did that on a couple of tests on `repo_test.go` you can take a look

Comment (noam-codefresh) [reply cmd/commands/common.go:81]: ok, I'll take a look on how to do that

**Review by roi-codefresh — APPROVED** [review]

**Pull request merged**